arduino-core/src/processing/app/packages/UserLibrary.java

```diff
@@ -148,12 +152,16 @@ public static UserLibrary create(UserLibraryFolder libFolderDesc) throws IOExcep
     }
 
     String declaredVersion = properties.get("version").trim();
-    Version version = VersionHelper.valueOf(declaredVersion);
+    Optional<Version> version = VersionHelper.valueOf(declaredVersion);
+    if (!version.isPresent()) {
+      System.out.println(
+          format(tr("Invalid version '{0}' for library in: {1}"), declaredVersion, libFolder.getAbsolutePath()));
+    }
 
     UserLibrary res = new UserLibrary();
     res.installedFolder = libFolder;
     res.name = properties.get("name").trim();
-    res.version = version.toString();
+    res.version = version.isPresent() ? version.get().toString() : declaredVersion;
```

Comment:
Could using "declaredVersion" have unintended consequences?  As I understand this, it would allow libraries to put any arbitrary string into Arduino's data structures.  Is everything else in the Arduino code able to handle any arbitrary text, when it's all been designed around the until-now strictly enforced major.minor.patch format?

Reply:
The `version` field for user libraries has always been a `String` and the format has never been forced to be semver compliant.

To compare version "strings" we use the [`VersionComparator`](https://github.com/arduino/Arduino/blob/fb5ed00547ae394784bee2bd9f617fdf4f54ebf5/arduino-core/src/cc/arduino/contributions/VersionComparator.java) class that compares two different version "strings", say A and B, using the following alogrithm:

1. parse A and B as semver
2. if both A and B are valid semver, compare them using semver rules
3. if A is valid semver but not B, then A is the greatest
4. if B is valid semver but not A, then B is the greatest
5. if both A and B are not valid semver, consider them equals

This logic allows libraires that are not semver compliant to soft transition to semver (since a valid semver version is always greater).

Reply:
Ok, sounds like you've thought this through quite carefully.  I was thinking simpler less-is-more in #7922, after having a user reports 1.8.6 unable to start up.